README.md

```diff
@@ -145,6 +153,163 @@ readData(dataInput) {
+
+The following is a membership listener registration by using `ClusterService.on()` function.
+
+```javascript
+client.clusterService.on('memberAdded', function (member) {
+    console.log('Member Added: The address is', member.address.toString());
+});
+```
+
+####  4.1.2. Listening for Distributed Object Events
+
+The Distributed Object Listener events are invoked when a distributed object is created and destroyed throughout the cluster. After the events, a listener callback function is called. Parameters of the function is as follows:
+
+`serviceName`: service name of the distributed object
+
+`name`: name of the distributed object
+
+`event`: name of the event that happened: either `created` or `destroyed`
+
+The structure of the callback function should be as follows.
+
+```javascript
+function listenerFunc(serviceName, name, event) {
+    // some operations
+}
+```
+
+The following is an example of adding a Distributed Object Listener.
+
+```javascript
+client.addDistributedObjectListener(function (serviceName, name, event) {
+    console.log('Distributed object event >>> ' + JSON.stringify({
+        serviceName: serviceName,
+        name: name,
+        event: event
+    }));
+}).then(function () {
+    var mapname = 'test';
+    // This causes a created event.
+    client.getMap(mapname);
+    // This causes no event because map was already created.
+    client.getMap(mapname);
+});
+```
+
+#### 4.1.3. Listening for Lifecycle Events
+
+The Lifecycle Listener notifies for the following events:
+- `starting`: A member is starting.
+- `started`: A member started.
+- `shuttingDown`: A member is shutting down.
+- `shutdown`: A member’s shutdown has completed.
+
+The following is an example of Lifecycle Listener that is added to config and its output.
+
+```javascript
+var clientConfig = new Config.ClientConfig();
+clientConfig.listeners.addLifecycleListener(function (state) {
+    console.log('Lifecycle Event >>> ' + state);
+});
+
+Client.newHazelcastClient(clientConfig).then(function (hazelcastClient) {
+    hazelcastClient.shutdown();
+});
+```
+
+**Output:**
+
+```
+Lifecycle Event >>> starting
+[DefaultLogger] INFO at ConnectionAuthenticator: Connection to 10.216.1.62:5701 authenticated
+[DefaultLogger] INFO at ClusterService: Members received.
+[ Member {
+    address: Address { host: '10.216.1.62', port: 5701, type: 4 },
+    uuid: 'dc001432-7868-4ced-9161-5649ff6f31fc',
+    isLiteMember: false,
+    attributes: {} } ]
+Lifecycle Event >>> started
+[DefaultLogger] INFO at HazelcastClient: Client started
+Lifecycle Event >>> shuttingDown
+Lifecycle Event >>> shutdown
+
+Process finished with exit code 0
+```
+
+### 4.2. Distributed Data Structure Events
+
+You can use listeners for distributed data structures. This part introduces how to use these listeners for each data structure.
+
+> **NOTE: Hazelcast Node.js Client is a TypeScript-based project but JavaScript does not have interfaces. Therefore, 
```

Comment:
I think this explains what a JS user should do when we say "implement X interface" very well and saves us from trouble of explaining it all the time. Very nice note, I like it 👍